File: Bwl.Framework/src/Logger/UI/LoggerForm.vb
Imports Bwl.Framework
Imports System.Windows.Forms

Public Class LoggerForm
    Private _logger As ILoggerDispatcher

    Public Sub New(logger As ILoggerDispatcher)
        InitializeComponent()

        _logger = logger
        _logger.ConnectWriter(_datagridLogWriter1)
    End Sub
End Class

File: Framework/Framework/Bwl.Framework.Windows/Logger/UI/LoggerForm.cs
using System.Runtime.Versioning;
using System.Windows.Forms;

namespace Bwl.Framework.Windows
{

    [SupportedOSPlatform("windows")]
    public partial class LoggerForm
    {
        private ILoggerDispatcher _logger;

        public LoggerForm(ILoggerDispatcher logger)
        {
            InitializeComponent();
            Application.EnableVisualStyles();

            _logger = logger;
            _logger.ConnectWriter(_datagridLogWriter1);
        }
    }
}